src/core/core.scale.js

```diff
@@ -296,10 +304,22 @@ var Scale = Element.extend({
 
 		// New implementations should return an array of objects but for BACKWARD COMPAT,
 		// we still support no return (`this.ticks` internally set by calling this method).
-		ticks = me.buildTicks() || [];
+		ticks = me.buildTicks();
 
 		// Allow modification of ticks in callback.
-		ticks = me.afterBuildTicks(ticks) || ticks;
+		if (ticks) {
+			ticks = me.afterBuildTicks(ticks);
+		} else {
+			// Support old implementations (that modified `this.ticks` directly in buildTicks)
+			me.ticks = me.afterBuildTicks(me.ticks);
```

Comment:
I'm not sure it's safe because previously `afterBuildTicks` was called with an empty array so if anyone reimplemented this (public) method to handle that case, it may not work anymore as expected.

Reply:
Fixed